plugin/trino-hive/src/main/java/io/trino/plugin/hive/orc/OrcPageSource.java

```diff
@@ -398,7 +403,7 @@ public UpdatedRowAdaptation(HiveUpdateProcessor updateProcessor, List<HiveColumn
         }
 
         @Override
-        public Block block(Page sourcePage, MaskDeletedRowsFunction maskDeletedRowsFunction, long filePosition)
+        public Block block(Page sourcePage, MaskDeletedRowsFunction maskDeletedRowsFunction, long filePosition, OptionalLong startRowId)
```

Comment:
Apart from `PositionAdaptation` where it is only checked for being empty, I don't see `startRowId` actually being used anywhere. Is this intentional?

Reply:
It's not, but I'd rather leave it in even if just for that isEmpty check. If that Adapter is used in a situation where the value is present it'll silently produce incorrect results.